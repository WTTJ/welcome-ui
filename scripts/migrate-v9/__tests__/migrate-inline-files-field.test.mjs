import { readFileSync } from 'fs'
import { resolve } from 'path'

import { describe, expect, it } from 'vitest'

import { setupMigrationTest } from '../helpers/test-setup.mjs'
import { migrateAll } from '../index.mjs'

describe('Inline Migration - Field Components', () => {
  const tempDir = setupMigrationTest('inline-field-test', '__fixtures__')

  it('migrates Field components with form props to Field with classNames', async () => {
    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'FieldComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('field-component.tsx')
  })
})
