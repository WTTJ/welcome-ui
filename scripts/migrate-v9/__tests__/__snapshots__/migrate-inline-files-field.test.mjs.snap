// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Field Components > migrates Field components with form props to Field with classNames > field-component.tsx 1`] = `
"import { Field } from 'welcome-ui/Field'
import { InputText } from 'welcome-ui/InputText'
import { Label } from 'welcome-ui/Label'

const register = (name: string, options?: object) => {
  return { name, ...options }
}

export const FieldComponent = () => {
  return (
    <form>
      <Field className="my-lg" error="my error message" label="Email" required>
        <Label className="items-center mb-sm"></Label>
        <InputText
          autoFocus
          dataTestId="wa-forgot-input-email"
          name="email"
          {...register('email', {
            pattern: 'This field has invalid format',
            required: 'This field is required',
            value: /fish/,
          })}
        />
      </Field>
    </form>
  )
}
"
`;
